src/utils/arcgis-auth.ts

```diff
@@ -0,0 +1,76 @@
+import { ArcGISIdentityManager } from '@esri/arcgis-rest-request';
+
+const ARCGIS_REST_USER_SESSION = '__ARCGIS_REST_USER_SESSION__';
+const ARCGIS_REST_USER_INFO = '__ARCGIS_REST_USER_INFO__';
+
+const updateSessionInfo = async (session?: ArcGISIdentityManager): Promise<string> => {
+  let email = '';
+  if (session) {
+    localStorage.setItem(ARCGIS_REST_USER_SESSION, session.serialize());
+    const user = await session.getUser();
+    email = user.email || '';
+    localStorage.setItem(ARCGIS_REST_USER_INFO, email);
+  } else {
+    localStorage.removeItem(ARCGIS_REST_USER_SESSION);
+    localStorage.removeItem(ARCGIS_REST_USER_INFO);
```

Comment:
As I can see, logout is a sync operation. If you use a separate sync function or request the user data before this function, you won't need to have a thunk in the slice.

Reply:
It's my fault. I forgot to destroy the session, which is async operation. It's added now, so we need a thunk that is not a problem I guess.